scripts/release/list_backports.py

```diff
@@ -0,0 +1,218 @@
+        for p in unfiltered_pulls:
+            if not p.merged:
+                # only consider merged backports
+                continue
+
+            if p.closed_at < self._start_date or p.closed_at >= self._end_date + timedelta(1):
+                # only concerned with PRs within time window
+                continue
+
+            if p.number in self._excludes:
+                # skip PRs that have been explicitly excluded
+                continue
+
+            # check for issues in this pr
+            issues_for_this_pr = {}
+            with io.StringIO(p.body) as buf:
+                for line in buf.readlines():
+                    line = line.strip()
+                    match = re.search(r"^Fixes[:]?\s*#([1-9][0-9]*).*", line)
+                    if not match:
+                        match = re.search(
+                            r"^Fixes[:]?\s*https://github\.com/zephyrproject-rtos/zephyr/issues/([1-9][0-9]*).*", line)
+                    if not match:
+                        continue
+                    issue_number = int(match[1])
+                    issue = self._repo.get_issue(issue_number)
+                    if not issue:
+                        raise ValueError(
+                            f'https://github.com/zephyrproject-rtos/zephyr/pull/{p.number} references issue {issue_number} which is not an actual issue!')
+
+                    issues_for_this_pr[issue_number] = issue
+
+            # report prs missing issues later
+            if len(issues_for_this_pr) == 0:
+                self._pulls_without_an_issue.append(p)
+                continue
+
+            pulls.append(p)
+
+            issue_map = issue_map | issues_for_this_pr
+
+        issues = list(issue_map.values())
+
+        # paginated_list.sort() does not exist
+        pulls = sorted(pulls, key=lambda x: x.number)
+        issues = sorted(issues, key=lambda x: x.number)
+
+        self._pulls = pulls
+        self._issues = issues
+
+        return self._pulls
+
+    def get_issues(self):
+        """Return GitHub issues fixed in the provided date window"""
+        if self._issues:
+            return self._issues
+
+        self.get_pulls()
+
+        return self._issues
+
+    def get_pulls_without_issues(self):
+        if self._pulls_without_an_issue:
+            return self._pulls_without_an_issue
+
+        self.get_pulls()
+
+        return self._pulls_without_an_issue
+
+
+def main():
+    args = parse_args()
+    bp = Backport(Github(args.token), args.base,
+                  args.start, args.end, set(args.excludes))
+
+    pulls_without_issues = bp.get_pulls_without_issues()
+    if pulls_without_issues:
+        s = 'Please ensure the body of each PR to a release branch contains "Fixes #1234"\n'
+        s += 'The following PRs are lacking associated issues:'
+        for p in pulls_without_issues:
+            s += f'\nhttps://github.com/zephyrproject-rtos/zephyr/pull/{p.number}'
+        raise ValueError(s)
```

Comment:
Would it make sense to use a print and explicit exit? It's a bit odd to see a backtrace when this kicks in but it may be just me... up to you really.